src/sentry/incidents/models/alert_rule.py

```diff
@@ -550,6 +550,20 @@ def build_handler(
         return self.trigger_action_class(action, incident, project)
 
 
+class FactoryRegistry:
+    def __init__(self) -> None:
+        self.by_action_service: dict[ActionService, ActionHandlerFactory] = {}
+        self.by_slug: dict[str, ActionHandlerFactory] = {}
```

Comment:
could we possibly add a comment about the difference between these two? when do we use one over the other perhaps?

Reply:
The only difference is which key you want to use for a lookup. If that isn't clear enough from the names, I'll add a comment.

Reply:
https://github.com/getsentry/sentry/pull/74936/commits/1f33e57458097c433a0c3ed0fc0d7c7fd2f5e718